Update docs for the simulation state dump/load methods (#1774)
NanoComp/meep: doc/docs/Python_User_Interface.md

             progress_interval=4,
             subpixel_tol=0.0001,
             subpixel_maxeval=100000,

>              loop_tile_base=0,

Reviewer: This looks a regression due to the rebase.

  the minimum refractive index (usually 1), and in practice $S$ should be slightly
  smaller.

+ **`loop_tile_base_db`, `loop_tile_base_eh` [`number`]** — To improve the [memory locality](https://en.wikipedia.org/wiki/Locality_of_reference)
  of the field updates, Meep has an experimental feature to "tile" the loops over the Yee grid
  voxels. The splitting of the update loops for step-curl and update-eh into tiles or subdomains
  involves a recursive-bisection method in which the base case for the number of voxels is
  specified using these two parameters, respectively. The default value is 0 or no tiling;
  a typical nonzero value to try would be 10000.

> 

Reviewer 2: Looks like a rebasing problem?

Author: Yeah. Please ignore this PR - Recreated this change in a  fresh branch
See https://github.com/NanoComp/meep/pull/1784

</div>

### Load and Dump Simulation State

The functions below add support for saving and restoring parts of the MEEP
Simulation state.

For all functions below, when dumping/loading state to/from a distributed filesystem
(using say, parallel HDF5) and running in a MPI environment, setting

> `single_parallel_file=True` will result in all processes writing/reading to/from

Reviewer 2: ```suggestion
`single_parallel_file=True` (the default) will result in all processes writing/reading to/from
```

sim2.run(...)
```

#### Load and Dump Fields

These functions can be used to dump (and later load) the raw fields, auxillary
fields and the dft fields at a certain timestamp. The timestamp at which the
dump happens is also saved so that simulation can continue from where it was saved.
The one pre-requisite of this feature is that it needs the Simulation object to have
been setup *exactly* the same as the one it was dumped from.

<a id="Simulation.dump_fields"></a>

<div class="class_members" markdown="1">

```python
def dump_fields(self, fname, single_parallel_file=True):
```

<div class="method_docstring" markdown="1">

Dumps the fields to the file `fname`.

</div>

</div>

<a id="Simulation.load_fields"></a>

<div class="class_members" markdown="1">

```python
def load_fields(self, fname, single_parallel_file=True):
```

<div class="method_docstring" markdown="1">

Loads a fields from the file `fname`.

</div>

</div>

#### Checkpoint / Restore.

The above dump/load related functions can be used to implement a
checkpoint/restore *like* feature for MEEP. The caveat of this feature is that
it does *not* store all the state required to re-create the `Simulation` object

> itself. Instead in expects the user to create and setup the new `Simulation`

Reviewer 2: ```suggestion
itself. Instead, it expects the user to create and set up the new `Simulation`
```
